src/CMakeLists.txt

```diff
@@ -1,8 +1,42 @@
-add_library(sqlpp-mysql
-		connection.cpp
-		prepared_statement.cpp
-		char_result.cpp
-		bind_result.cpp
-		detail/connection_handle.cpp)
-
-install(TARGETS sqlpp-mysql DESTINATION lib)
+
+add_library(sqlpp-mysql SHARED)
+add_library(sqlpp11::mysql ALIAS sqlpp-mysql)
+
+add_library(sqlpp-mysql-static STATIC)
+add_library(sqlpp11::mysql::static ALIAS sqlpp-mysql-static)
```

Comment:
I think we should stick with add_library(sqlpp-mysql). The usual recommendation is to leave the choice up to the developer. CMake provides a variable for this the developer can set. 

For example on the command line you can specify: 
`cmake -B build -DBUILD_SHARED_LIBS=YES`

This will build every lib (which left the choice open) as shared. 

If you as a developer want just this lib shared, you can easily overwrite in in your application (This is the reason why I personally recommend putting every dependencies in it's own directory / scope, this allows to set this variable before calling FetchContent_MakeAvailable() without affecting any other lib). 

I think we should stick to this recommend approach of letting the developer decide by passing a variable. It's consistent with other lib (at least those, who know of this convention) and you don't have to repeat yourself here everywhere.